File: web/src/api/resource.ts
import request from '../utils/request'

export interface Classroom {
  id?: number
  name: string
  capacity: number
  equipment: string
  status: 'available' | 'maintenance' | 'closed'
}

export const getClassrooms = () => {
  return request.get<{ data: Classroom[] }>('/resources/classrooms')
}

export const createClassroom = (data: Classroom) => {
  return request.post('/resources/classrooms', data)
}

export const updateClassroom = (id: number, data: Classroom) => {
  return request.put(`/resources/classrooms/${id}`, data)
}

export const deleteClassroom = (id: number) => {
  return request.delete(`/resources/classrooms/${id}`)
}

File: api/models/resource.go
package models

import "time"

type Classroom struct {
	ID        uint      `gorm:"primaryKey" json:"id"`
	Name      string    `gorm:"size:50;not null" json:"name"`
	Capacity  int       `gorm:"default:30" json:"capacity"`
	Equipment string    `gorm:"type:text" json:"equipment"`
	Status    string    `gorm:"type:enum('available','maintenance','closed');default:'available'" json:"status"`
	CreatedAt time.Time `json:"created_at"`
}
